### PR TITLE
Clean up the ServerInstance class and remove the un-necessary ip address stored

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
- Remove the un-necessary IP address field
- Remove the error log for the UnknownHostException
- Always use the cached host name and short host name if exists
- This can reduce lots of errors in tests as host passed in might not exist, e.g. "testHost"

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/response/ServerInstance.java [hunk 1]
```diff
@@ -140,8 +110,12 @@ public int getPort() {
     return _port;
   }
 
-  public InetAddress getIpAddress() {
-    return _ipAddress;
+  public ServerInstance withSeq(int seq) {
+    return new ServerInstance(_hostname, _port, seq, _shortHostName);
+  }
+
+  public String getShortHostName() {
```
Comment (jackjlli) [inline pinot-common/src/main/java/org/apache/pinot/common/response/ServerInstance.java]: It'd be good to put all the getter methods together.

File: pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java [hunk 2]
```diff
@@ -18,61 +18,27 @@
...
+    assertNotEquals(new ServerInstance("localhost", 8081), new ServerInstance("localhost", 8082));
 
-      ServerInstance instance1 = new ServerInstance("127.0.0.1", 8080);
-      Assert.assertEquals(instance1.getPort(), 8080, "Port check");
-      Assert.assertEquals(instance1.getPort(), 8080, "Hostname check");
-      Assert.assertEquals(instance1.getHostname(), ipAddr.getHostName(), "Host check");
-      Assert.assertEquals(instance1.getIpAddress(), ipAddr, "IP check");
-    }
+    // Same port but different host
+    assertNotEquals(new ServerInstance("foo", 8080), new ServerInstance("bar", 8080));
```
Comment (jackjlli) [inline pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java]: Can we include a test to test the case when host name is an ip address, like `127.0.0.1`?

File: pinot-common/src/main/java/org/apache/pinot/common/response/ServerInstance.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/common/response/ServerInstance.java]: Done

**Comment by mcvsubbu** [comment]
I think the IP addresses are there for a reason. Perhaps azure/kubernetes clusters? I remember some discussion around it.

**Comment by mcvsubbu** [comment]
Or, maybe it was in the context of caching the IP address while using connection pools. Are we totally disabled on connection pools now?

File: pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java]: Done

**Comment by Jackie-Jiang** [comment]
@mcvsubbu We never fetch the ip address from the server instance. The reason why we have the ip address is to resolve "127.0.0.1" to "localhost" so they are considered the same host. That is not necessary because the server instance is always derived from the instance name

File: pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java (hunk 2 above)
Comment (jackjlli) [reply pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java]: Thanks. How about the test comparing with `127.0.0.1` as hostname and `localhost` as hostname? Would they be the same? Can we include this test as well?

Comment (Jackie-Jiang) [reply pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java]: That won't be the same as we do not resolve the ip address

**Comment by Jackie-Jiang** [comment]
@mcvsubbu Actually the domain name expects the host name instead of the ip address, so I'll add back the ip address to host lookup

File: pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-transport/src/test/java/org/apache/pinot/transport/common/ServerInstanceTest.java]: Added back the IP address lookup and added test

**Pull request merged**